src/types/basic/uint.ts

```diff
@@ -18,9 +18,6 @@ export class UintType<T> extends BasicType<T> {
 
 export class NumberUintType extends UintType<number> {
   assertValidValue(value: unknown): asserts value is number {
-    if (!(Number.isSafeInteger(value as number) || value === Infinity)) {
```

Comment:
We still need this check in assertValidValue, we still only allow safe integers and infinity

Reply:
And actually, we're missing a byte length check here. eg: that `value < (BigInt(2) ** this.byteLength)`